src/compiler/crystal/semantic/semantic_visitor.cr

```diff
@@ -68,6 +68,24 @@ abstract class Crystal::SemanticVisitor < Crystal::Visitor
     node.expanded = expanded
     node.bind_to(expanded)
     false
+  rescue ex : CrystalPath::NotFoundError
```

Comment:
Can we move this rescue just around `find_in_path` because that's where the error can happen? This makes it much clearer when the exception is expected to happen.

Reply:
I considered that, but the problem is that the exception raised there would subsequently be rescued in line 89. I don't think it makes sense to make that rescue handler work around that.
A slightly less confusing option could be to create the error in a rescue around `find_in_path`, assign it to a local variable and only raise in an `else` rescue handler. Not sure if that's worth it, though.

Reply:
Oh, I see. Yeah, let's leave it like that for now.

Reply:
That said, I don't understand why `CrystalPath::NotFoundError` is handled in a special way here, instead of just being a regular Crystal::Exception and having a sensible `to_s`.

Reply:
The idea was to have `CrystalPath::NotFoundError` as a plain internal error type, not related to any user code interpretation. `CrystalPath` should not care wheter it was invoked for a `require` or something else. It also doesn't know the code location, so it can't really create a proper user-facing error.
`NotFoundError` just communicates that finding a path failed. What that means depends on the context. In the context of interpreting a `Require` node it means the require has failed and an appropriate error is raised for that specific situation.